state/txindex/indexer.go

```diff
@@ -36,6 +36,11 @@ type TxIndexer interface {
 	GetRetainHeight() (int64, error)
 
 	SetRetainHeight(retainHeight int64) error
+
+	// Close closes the underlying database that a TxIndexer uses. It is the
```

Comment:
```suggestion
	// Close frees any resources associated with the indexer. Note that calls to other methods after calling Close may lead to panic.
```

`TxIndexer` had no notion of a database up until this point. The message should somehow be amended to introduce this gradually.